weather.py

```diff
@@ -0,0 +1,306 @@
+# Imports written in alphabetical order
+from os import path
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_pre = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # Dictionaries as class instances to store calculation results
+    e_result = dict()
+    a_result = dict()
+    c_result = dict()
+
+    # Report Number to print
+    report_num = 1
+
+    # Initialize Lists in these indexes of dictionary
+    c_result["max_temp"] = []
+    c_result["min_temp"] = []
+
+    # List with months to access by numerical form
+    months = ["Jan", "Feb", "Mar",
+              "Apr", "May", "Jun",
+              "Jul", "Aug", "Sep",
+              "Oct", "Nov", "Dec"
+              ]
+
+    # Class Constructor
+    def __init__(self, path, e, a, c, b):
+        self.__path = path
+        self.__e = e
+        self.__a = a
+        self.__c = c
+        self.__b = b
+
+        # Variables used to cater multiple inputs of a specific argument
+        self.__curr_e = None
+        self.__curr_a = None
+        self.__curr_c = None
+
+    # Method to extract weather files to specified path
+    def Extract(self):
+        # Extract the weatherman.zip file to path
+        with zipfile.ZipFile("weatherfiles.zip", 'r') as zip_ref:
+            zip_ref.extractall(self.__path)
+
+    # Method to get data to print from class level data structures according to mode passed
+    def getData(self, mode):
+        if mode == "e":
+
+            high_date = Weatherman.e_result['maxdate']
+            low_date = Weatherman.e_result['mindate']
+            high_temp = Weatherman.e_result['maxtemp']
+            low_temp = abs(Weatherman.e_result['mintemp'])
+
+            high_date = high_date.split("-")  # Separating date into day, month and year
+            low_date = low_date.split("-")
+
+            high_month = Weatherman.months[int(high_date[1])-1]
+            high_year = high_date[2]
+            low_month = Weatherman.months[int(low_date[1])-1]
+            low_year = low_date[2]
+
+            return high_temp, low_temp, high_month, low_month, high_year, low_year
+
+        elif mode == "a":
+            high_avg_temp = round(Weatherman.a_result['avgmaxtemp'])
+            low_avg_temp = round(abs(Weatherman.a_result['avgmintemp']))  # abs used to counteract negative results
+            mean_avg_hum = round(Weatherman.a_result['avghumid'])
+
+            return high_avg_temp, low_avg_temp, mean_avg_hum
+
+        elif mode == "c":
+            max_pads = []
+            min_pads = []
+            max_pad = ""
+            min_pad = ""
+
+            max_temp = Weatherman.c_result["max_temp"]
+            min_temp = Weatherman.c_result["min_temp"]
+
+            # Run loop for all the days logged
+            for i in range(len(max_temp)):
```

Comment:
Instead of writing generic and bigger methods, it's better to write smaller and specific methods. Extract each `if` block into a separate method and directly call them, do the same thing with all functions where you have separated logic using the `if` statement based on mode.

Reply:
change made as suggested.